### DISCUSSION TITLE
Reusing restored volumes in new helm deployment

### DISCUSSION THREAD

**Discussion by Hr46ph** [longhorn/longhorn#7895] [Q&A]
I realize this isn't a Longhorn specific question and perhaps a lack of knowledge on my end. Still I hope perhaps someone can help me out as Im struggling with this and I cant figure it out.

I am running Longhorn for storage provider. I installed Vaultwarden via helm and supplied values.yaml in which I set a domain name and custom size for data and files volumes. Thats it.

Vaultwarden runs fine, and after creating an ingress I can access the gui. I create a user and import a json with some passwords.

I create a backup of the volumes with Longhorn.
I delete vaultwarden via helm uninstall
I delete the current pv's and pvc's
restore pv's from backup.
I create vaultwarden namespace
create pvc vaultwarden-files and vaultwarden-data for the restored volumes.

I reinstall vaultwarden via helm and I kind of expect it to reuse the pvc's as they exist with the same name it would have created them. Instead, it insists on recreating new pv's and pvc's with that name and append the pod name 'vaultwarden-0' to it.

At this point, no matter what I try I cannot get the pod or stateful set to reuse the restored volumes.

I realize I might lack some knowledge on Kubernetes as I am still learning. But after a few hours struggling with this I figured I'd ask because from what I understand it shouldn't be difficult.

I also tried to edit the newly created pvc's to point to the restored pv's. It didnt accept.
I tried to edit the statefulset with kubectl edit, it accepts but it doesnt' change anything. Reopening shows the original values again.

Appreciate some help here!
Thanks!

**Comment by DamiaSan** [comment]
It seems to be a problem of vaultwarden which doesn't check for the existence of the PVC and create always a new one. Maybe it could be better to ask directly on vaultwarden github page?

**Reply by Hr46ph** [reply]
Thanks, yeh I did:
https://github.com/guerzon/vaultwarden/discussions/71

No reply yet, which led me to ask here as well.

So, for situations like these, how else would I backup and restore? I see two other options:
1. velero
2. sidecar sqlite3 to another persistent volume or nfs storage.

I tried 1 as well, but so far I have not been lucky in restoring. Same issue as with the longhorn backup & restore.

Option 2 I have not tried yet. It might actually be the better option since that way I can take hot backups of sqlite database, while snapshots and volume backups might leave the database in an inconsistent state.
